js/components/addEditBookmark.js

```diff
@@ -23,7 +23,17 @@ class AddEditBookmark extends ImmutableComponent {
     this.onClick = this.onClick.bind(this)
     this.onSave = this.onSave.bind(this)
     this.onRemoveBookmark = this.onRemoveBookmark.bind(this)
+    this.state = {
+      bookmarkNameValid: true,
+      saveButtonDisabled: true
+    };
   }
+
+  shouldComponentUpdate (nextProps, nextState) {
+    return (Object.keys(nextState).some((prop) => nextState[prop] !== this.state[prop]) || 
+      Object.keys(nextProps).some((prop) => nextProps[prop] !== this.props[prop]))
+  }
```

Comment:
And remove this `shouldComponentUpdate`, no immutable component should need to ever override it.